rest-api-spec/src/main/resources/rest-api-spec/test/indices.simulate_index_template/10_basic.yml

```diff
@@ -0,0 +1,166 @@
+---
+"Simulate index template without new template in the body":
+  - skip:
+      version: " - 7.99.99"
+      reason: "simulate index template API has not been backported"
+      features: allowed_warnings
+
+  - do:
+      allowed_warnings:
+        - "index template [test] has index patterns [te*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [test] will take precedence during new index creation"
+      indices.put_index_template:
+        name: test
+        body:
+          index_patterns: te*
+          template:
+            settings:
+              number_of_shards:   1
+              number_of_replicas: 0
+            mappings:
+              properties:
+                field:
+                  type: keyword
+
+  - do:
+      indices.simulate_index_template:
+        name: test
+
+  - match: {template.settings.index.number_of_shards: "1"}
+  - match: {template.settings.index.number_of_replicas: "0"}
+  - match: {template.mappings._doc.properties.field.type: "keyword"}
+  - match: {overlapping: {}}
+
+---
+"Simulate index template specifying a new template":
```

Comment:
Can you add a test that uses `composed_of` so we can ensure that resolving the component template settings works correctly?

Reply:
Done, thanks for the suggestion